databroker/assets/base_registry.py

```diff
@@ -463,6 +463,8 @@ def register_resource(self, spec, root, rpath, rkwargs,
             The uid of the created resource.
 
         '''
+        if callable(getattr(self, '_create_resource_index', None)):
```

Comment:
If this exists but is not a callable, what does that mean? Should it pass silently?

Reply:
I agree, `hasattr` or `... is not None` is what I would have naively expected to be here.